docs/tests-mvp.md

```diff
@@ -0,0 +1,402 @@
+- (Minimize) The bandwidth consumed to fetch a file. Lower is Better
+  - To compute this, capture: The number of duplicated blocks received vs. total block count
+- (Reach) The number of nodes that were able to fetch all files as instructed. (Reach 100% success of all fetches)
+- (Reach) No node is expected to crash/panic during this Test Plan. (Reach 0% crashes)
+
+
+### Parameters
+
+- **Network Parameters**
+  - `Region` - Region or Regions where the test should be run at (default to single region)
+  - Ran with with an arbitraty amount of nodes (from 10 to 1000000) - N
+- **Image Parameters**
+  - Number of nodes with full replica of the dataset initially (from 1 to 10) - M
+  - Ran with custom libp2p & IPFS suites (swap in/out Bitswap & GraphSync versions, Crypto Channels, Transports and other libp2p components)
+
+This test is not expected to support:
+
+- An heterogeneus network in which nodes have different configurations
+
+### Test Narratives
+
+#### `Test:` _NAME_
+
+- **Warm up**
+  - Boot N nodes
+  - Connect each node to the node next to it (hash ring)
+  - Run multiple DHT random-walk queries to populate the finger tables
+  - Run a discovery service provided by Redis (to ensure that every node keeps getting at least one another node to connect)
+  - Load the datasets in M nodes
+- **Act I** - Access Wikipedia
+  - %W of the nodes access Wikipedia
+- **Act II** - Access NPM
+  - %W of the nodes access NPM
+- **Act III** - Access ImageNet
+  - %W of the nodes access ImageNet
+
+## `Plan:` Nodes Connectivity (Transports, Hole Punching, Relay)
+
+Ensuring that a node can always connect to the rest of the network, if not completely isolated.
+
+### What is being optimized (min/max, reach)
+
+- (Reach) The number of nodes that are able to dial to any other node (100%)
+
+### Parameters
+
+- **Network Parameters**
+  - `Region` - Region or Regions where the test should be run at (default to single region)
+  - Ran with with an arbitraty amount of nodes (from 10 to 1000000) - N
+  - Nodes being beyind a NAT/Firewall - F (F is a % of N)
+  - Nodes running the Image with IPFS on a Browser - B (B is a % of N)
+  - Nodes running the Image with go-ipfs using only TCP - T (T is a % of N)
+  - Nodes running the Image with go-ipfs using only QUIC - Q (Q is a % of N)
+  - Nodes running the Image with go-ipfs using only WebSockets - W (W is a % of N)
+  - Nodes running the Image with go-ipfs using only WebRTC - C (C is a % of N)
+- **Image Parameters**
+  - Image A - Base `go-ipfs`
+    - `Transport` The only transport to be used
+  - Image B - Base `js-ipfs` running in a Browser
+    - `Browser` The Browser in which js-ipfs will be running from
+
+### Test Narratives
+
+#### `Test:` _NAME_
+
+- **Warm up**
+  - Create the Bootstrapper nodes that are connected among themselves and support every transport
+- **Act I**
+  -
+- **Act II**
+  -
+- **Act III**
+  -
+
+## `Plan:` Providing Content (Content Routing / DHT)
```

Comment:
@daviddias Before I forget to comment, just wanted to point out that IMHO the biggest change we'll see in the DHT (until we have a message oriented DHT) will come from removing the clients behind NATs. The idea of having nodes remove themselves from the network based on information they receive from AutoNAT should hopefully do the job, but this change has the serious ability to make/break the network.

This is obviously a separate logical issue from testing DHT query resolution, but no matter where we decide to test it I think it's important we do so explicitly.

Reply:
Thanks! There is a Plan that is intended to capture the NAT challenges well. We will continue iterating on the Narratives as well.